OpenRA.Game/SelectableExts.cs

```diff
@@ -31,7 +31,7 @@ public static int SelectionPriority(this ActorInfo a)
 		public static int SelectionPriority(this Actor a)
 		{
 			var basePriority = a.Info.TraitInfo<SelectableInfo>().Priority;
-			var lp = a.World.LocalPlayer;
+			var lp = a.World.RenderPlayer;
```

Comment:
Please change this to `var viewer = a.World.RenderPlayer ?? a.World.LocalPlayer`,  and then update the `lp` references below.  The fallback to LocalPlayer is needed for players who enable the visibility cheat (which sets RenderPlayer null).

Reply:
The fallback had to be a little different:
```
var viewer = World.LocalPlayer.Spectating ? World.RenderPlayer : World.LocalPlayer;
```

That's because `RenderPlayer` is `null` when spectating in "Disable shroud" mode and the fallback in that case was breaking the feature.

Reply:
It actually had to be:
```
var viewer = (World.LocalPlayer == null || World.LocalPlayer.Spectating) ? World.RenderPlayer : World.LocalPlayer;
```

To account for spectators that are not players.

Reply:
I'm confused by
> That's because `RenderPlayer` is `null` when spectating in "Disable shroud" mode and the fallback in that case was breaking the feature. 

`a.World.RenderPlayer ?? a.World.LocalPlayer` means "RenderPlayer, unless its null, in which case use LocalPlayer". Spectators in the "Disable shroud" mode should have both `RenderPlayer` and `LocalPlayer` null, so it shouldn't matter which of the two nulls it uses?

We use this pattern in 13 other places already, so if there is a case where this fails then this will need to be fixed in all of them (but not here!).

Reply:
Spectators that used to be players have only `RenderPlayer` as `null` while `LocalPlayer` is still the player they used to be while playing.

Reply:
I mean - the fallback to `LocalPlayer` was breaking the feature in "Disable Shroud" mode for players that have resigned. I don't know if that's a problem in the other places where this pattern is used.

Reply:
`var viewer = a.World.RenderPlayer ?? a.World.LocalPlayer` does not do the trick here. I've tried and it breaks the feature for defeated players that are spectating because `LocalPlayer` in that case is not null.